pkg/network/common/validation.go

```diff
@@ -147,12 +147,18 @@ func ValidateHostSubnet(hs *networkapi.HostSubnet) error {
 	for i, egressIP := range hs.EgressIPs {
 		if _, err := validateIPv4(string(egressIP)); err != nil {
 			allErrs = append(allErrs, field.Invalid(field.NewPath("egressIPs").Index(i), egressIP, err.Error()))
+			// if any one of the IPs is invalid, we reset the whole array
+			hs.EgressIPs = nil
+			break
 		}
 	}
 
 	for i, egressCIDR := range hs.EgressCIDRs {
 		if _, err := validateCIDRv4(string(egressCIDR)); err != nil {
 			allErrs = append(allErrs, field.Invalid(field.NewPath("egressCIDRs").Index(i), egressCIDR, err.Error()))
+			// if any one of the CIDRs is invalid, we reset the whole array
+			hs.EgressCIDRs = nil
```

Comment:
I'm not familiar enough to have an opinion on this, is not too disruptive to reset everything, maybe filtering out the bad ones?
Anyway, validation does not seems the right place to do this

Reply:
> I'm not familiar enough to have an opinion on this, is not too disruptive to reset everything, maybe filtering out the bad ones?

I'm not familiar either, but the first thing I did was to reset just the one incorrect value, but because of the way we check for CIDR patterns, I cannot leave an empty string and wipe out just one CIDR. I'd have to leave the whole field empty.

> Anyway, validation does not seems the right place to do this

I agree that this is not the right place, but if I were to do this in the subnets file, I'd have to do the check again to specifically see if its egressCIDRs or IPs that are failing. I was hoping for more ideas in the review.

Reply:
well, kube-proxy filters out the invalid IP addresses for some things
https://github.com/aojea/kubernetes/blob/90918a0a873f068b8be5d1e08f7eee8054d9c36e/pkg/proxy/service.go#L157-L184

it may be a possibility, filter the bad ones and log a warning in `pkg/network/master/subnets.go`

Reply:
hmm ok let me see if I push this back to `subnets.go` and stop making `validation.go` ugly

Reply:
> is not too disruptive to reset everything, maybe filtering out the bad ones?
Yes. you may define several egress cidrs per hostsubnet, reseting everything can provoke downtime in the whole namespace.

Reply:
It would be consistent with our behavior in other situations to just block all egress from the namespace in this case

Reply:
> It would be consistent with our behavior in other situations to just block all egress from the namespace in this case

hmm so I'll continue to just erase all the egress then.

Reply:
> It would be consistent with our behavior in other situations to just block all egress from the namespace in this case

 OK that's fair enough